### DISCUSSION TITLE
few performance ideas

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [uNetworking/uWebSockets#1434] [Ideas]
- For BoringSSL, have you looked at if using [`TLS_with_buffers_method`](https://boringssl.googlesource.com/boringssl/+/HEAD/PORTING.md#crypto_buffer) has any performance improvements over [`TLS_METHOD`](https://cs.github.com/uNetworking/uSockets/blob/15b5036d88a907fc3c386f1de751d8e2fdc6d379/src/quic.c?q=TLS_method#L582) + [`TLS_METHOD`](https://cs.github.com/uNetworking/uSockets/blob/15b5036d88a907fc3c386f1de751d8e2fdc6d379/src/crypto/openssl.c?q=TLS_method#L431)?
- For HTTP/1.1 parsing, have you looked at if using [picohttpparser](https://github.com/h2o/picohttpparser) improves the performance? When running http_load_test, much of the time is spent on reading http headers and that's something picohttpparser does very quickly because it uses SIMD
- For async DNS, have you looked at the very old [CFNetworking](https://developer.apple.com/library/archive/documentation/NetworkingInternet/Conceptual/NetworkingTopics/Articles/ResolvingDNSHostnames.html) APIs for this? The Networking.framework APIs are newer but I think would require linking more objective-c frameworks and that can cause boot performance to suffer

**Comment by ghost** [comment]
CRYPTO_BUFFER sounds very interesting - lowering memory usage per SSL connection is welcome.

Re picohttpparser - we have a similar approach with SIMD usage already. Re DNS - we cannot import dependencies like that and there is no need, we just need to put that blocking lookup in a thread and hide it for the user.
